internal/fromproto6/importresourcestate.go

```diff
@@ -0,0 +1,47 @@
+package fromproto6
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/internal/fwserver"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+// ImportResourceStateRequest returns the *fwserver.ImportResourceStateRequest
+// equivalent of a *tfprotov6.ImportResourceStateRequest.
+func ImportResourceStateRequest(ctx context.Context, proto6 *tfprotov6.ImportResourceStateRequest, resourceType tfsdk.ResourceType, resourceSchema *tfsdk.Schema) (*fwserver.ImportResourceStateRequest, diag.Diagnostics) {
+	if proto6 == nil {
+		return nil, nil
+	}
+
+	var diags diag.Diagnostics
+
+	// Panic prevention here to simplify the calling implementations.
+	// This should not happen, but just in case.
+	if resourceSchema == nil {
+		diags.AddError(
+			"Unable to Create Empty State",
+			"An unexpected error was encountered when creating the empty state. "+
+				"This is always an issue in the Terraform Provider SDK used to implement the provider and should be reported to the provider developers.\n\n"+
```

Comment:
Just wondering about wording here. Do we want to refer to `SDK` or might this cause confusion when Framework is being used?

Reply:
Created a followup issue for discussion: https://github.com/hashicorp/terraform-plugin-framework/issues/338